modules/nf-core/glimpse2/chunk/main.nf

```diff
@@ -0,0 +1,51 @@
+process GLIMPSE2_CHUNK {
+    tag "$meta.id"
+    label 'process_low'
+
+    beforeScript  """
+        if cat /proc/cpuinfo | grep avx2 -q
+        then
+            echo "Feature AVX2 present"
+        else
+            echo "Feature AVX2 not present on node"
+            exit 1
+        fi
+    """
+    conda "bioconda::glimpse-bio=2.0.0"
+    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
+        'https://depot.galaxyproject.org/singularity/glimpse-bio:2.0.0--hf340a29_0':
+        'quay.io/biocontainers/glimpse-bio:2.0.0--hf340a29_0' }"
+
+    input:
+    tuple val(meta), path(input), path(input_index), val(region), path(map)
+    val(model)
+
+    output:
+    tuple val(meta), path("*.txt"), emit: chunk_chr
+    path "versions.yml"           , emit: versions
+
+    when:
+    task.ext.when == null || task.ext.when
+
+    script:
+    def prefix    = task.ext.prefix ?: "${meta.id}"
+    def model_cmd = model=="rec" ? "--recursive" : model=="seq" ? "--sequential" : model=="unv" ? "--uniform-number-variants" : ""
```

Comment:
Is there a reason there's an alias used for the model? 

If the value of `mode` was just 'recursive', 'sequential' or 'uniform-number-variants', you could pass directly in the command section like:

```
--${model}
```

You could lean on validation in the workflow schema to limit the options available.

Reply:
You are completely right :)
When you talk about the workflow schema to limit the options available, is it about the different pattern available in the meta.yml ?

Reply:
So, when people write workflows that use this module, and if they provide an option to pass through to this parameter, they can restrict the available values like [this](https://github.com/nf-core/differentialabundance/blob/c5de0561506f73887979909274b9f7155ef1cf90/nextflow_schema.json#L27). 

A pattern in the meta.yml would be a very sensible thing to do for documentation anyway, but I'm not sure how enforced they are. You could add a different bit of groovy to the module where you had your code before to check that `model` is one of the allowed values, but I haven't seen a lot of that in module code and it's probably easier to let the tool die if it received a bad parameter name.